.github/workflows/client.yml

```
@@ -0,0 +1,49 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows
+on:
+  push:
+    branches:
+      # TODO: Run on master after we're fully migrated from Circle CI
+      # - master
+      - "rfc-18/**"
+  pull_request:
+      branches:
+      # TODO: Run on all branches after we're fully migrated from Circle CI
+      - "rfc-18/**"
```

If we decide to remove Circle CI config we can switch to the final branches setup here.
```suggestion
  push:
    branches:
      - master
  pull_request:
```